x-pack/plugins/fleet/server/services/secrets.ts

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ElasticsearchClient } from '@kbn/core/server';
+import type { BulkResponse, DeleteResponse } from '@elastic/elasticsearch/lib/api/types';
+
+import { keyBy } from 'lodash';
+import { set } from '@kbn/safer-lodash-set';
+
+import { packageHasNoPolicyTemplates } from '../../common/services/policy_template';
+
+import type {
+  NewPackagePolicy,
+  PackagePolicyConfigRecordEntry,
+  RegistryStream,
+} from '../../common';
+
+import {
+  doesPackageHaveIntegrations,
+  getNormalizedDataStreams,
+  getNormalizedInputs,
+} from '../../common/services';
+
+import type {
+  PackageInfo,
+  PackagePolicy,
+  RegistryVarsEntry,
+  Secret,
+  VarSecretReference,
+  PolicySecretReference,
+} from '../types';
+
+import { FleetError } from '../errors';
+import { SECRETS_INDEX } from '../constants';
+
+import { appContextService } from './app_context';
+
+interface SecretPath {
+  path: string;
+  value: PackagePolicyConfigRecordEntry;
+}
+
+// This will be removed once the secrets index PR is merged into elasticsearch
+function getSecretsIndex() {
+  const testIndex = appContextService.getConfig()?.developer?.testSecretsIndex;
+  if (testIndex) {
+    return testIndex;
+  }
+  return SECRETS_INDEX;
+}
+
+// TODO: QUESTION do we need audit logging in this service?
```

Comment:
@kpollich Will we want auditing when a secret is created in the .secrets index? I wasn't sure as the user doesn't directly create them it's more a side effect of package policy creation

Reply:
Yeah I think it'd be good have audit logging when documents are added to this index. Should be a fairly quick addition I hope.

Reply:
Great thanks I'll do that now 👍